src/main/java/fi/metatavu/metamind/bot/functions/BeerApiBeerInfoMetaBotFunction.java

```diff
@@ -0,0 +1,164 @@
+    String flavorsVariable = functionParams.get("flavors");
+    String imagesVariable = functionParams.get("images");
+    String untappdRatingVariable = functionParams.get("untappdRating");
+    String rateBeerOverallScoreVariable = functionParams.get("rateBeerOverallScore");
+    String rateBeerStyleScoreVariable = functionParams.get("rateBeerStyleScore");
+
+    try {
+      ApiClient apiClient = getApiClient(getMetamindStory(session));
+      BarBeersApi barBeersApi = apiClient.buildClient(BarBeersApi.class);
+      Beer beer = barBeersApi.findBeer(barId, NumberUtils.createLong(beerId));
+      if (beer == null) {
+        logger.error("Cannot find beer with barId {} and beerId {}", barId, beerId);
+        return;
+      }
+
+      if (nameVariable != null && beer.getName() != null) {
+        story.getVariablesState().set(nameVariable, beer.getName());
+      }
+
+      if (descriptionVariable != null && beer.getDescription() != null) {
+        story.getVariablesState().set(descriptionVariable, beer.getDescription());
+      }
+
+      if (abvVariable != null && beer.getAbv() != null) {
+        story.getVariablesState().set(abvVariable, beer.getAbv());
+      }
+
+      if (ibuVariable != null && beer.getIbu() != null) {
+        story.getVariablesState().set(ibuVariable, formatDouble(beer.getIbu()));
+      }
+
+      if (styleVariable != null && beer.getStyle() != null) {
+        story.getVariablesState().set(styleVariable, beer.getStyle());
+      }
+
+      if (breweryVariable != null && beer.getBrewery() != null) {
+        story.getVariablesState().set(breweryVariable, beer.getBrewery());
+      }
+
+      if (caloriesVariable != null && beer.getCalories() != null) {
+        story.getVariablesState().set(caloriesVariable, beer.getCalories());
+      }
+
+      if (flavorsVariable != null && beer.getFlavors() != null && !beer.getFlavors().isEmpty()) {
+        story.getVariablesState().set(flavorsVariable, String.join(", ", beer.getFlavors()));
+      }
+
+      if (imagesVariable != null && beer.getImages() != null && !beer.getImages().isEmpty()) {
+        story.getVariablesState().set(imagesVariable, beer.getImages().stream().map(BeerImage::getMedium).collect(Collectors.joining("|")));
+      }
+
+      if (untappdRatingVariable != null && beer.getUntappdRating() != null) {
+        story.getVariablesState().set(untappdRatingVariable, formatBeerRating(beer.getUntappdRating()));
+      }
+
+      if (rateBeerOverallScoreVariable != null && beer.getRateBeerOverallScore() != null) {
+        story.getVariablesState().set(rateBeerOverallScoreVariable, formatBeerRating(beer.getRateBeerOverallScore()));
+      }
+
+      if (rateBeerStyleScoreVariable != null && beer.getRateBeerStyleScore() != null) {
+        story.getVariablesState().set(rateBeerStyleScoreVariable, formatBeerRating(beer.getRateBeerStyleScore()));
+      }
+
+    } catch (Exception e) {
+      logger.error("Could not set variable states", e);
+      return;
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this redundant jump. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3626)